tools/replicate_server_settings.rb

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env ruby
+require File.expand_path("../config/environment", __dir__)
+require 'trollop'
+
+opts = Trollop.options(ARGV) do
+  banner "USAGE:   #{__FILE__} -s <server id> -p <path/to/the/settings> \n" \
+         "Example: #{__FILE__} -d -s 1 -p ems/ems_amazon/additional_instance_types"
+
+  opt :dry_run,  "Dry Run",                                                :short => "d"
+  opt :serverid, "Replicating source server Id (default: current server)", :short => "s"
+  opt :path,     "Replicating source path within advanced settings hash",  :short => "p", :default => ""
+end
+
+puts opts.inspect
+Trollop.die :path, "is required" unless opts[:path_given]
+
+def replicate(opts)
```

Comment:
I think this would be better not using an options hash.

Then you could collapse the first line into a default value:

```ruby
def replicate(path_string, dry_run, server = MiqServer.my_server)
  ...
end
```

Clearly the caller of `replicate` would have to change also (to look up the server if given an id, but I don't think that's so bad. Or you could put the "find the source server" logic outside of the replicate method entirely.

Reply:
done